ndsplines/ndsplines.py

```diff
@@ -91,79 +102,67 @@ def compute_basis_coefficient_selector(self, x, nus=0):
                 nu = nus[i]
             if self.periodic[i]:
                 n = t.size - k - 1
-                x[i,:] = t[k] + (x[i,:] - t[k]) % (t[n] - t[k])
+                x[:,i] = t[k] + (x[:,i] - t[k]) % (t[n] - t[k])
                 extrapolate_flag = False
             else:
                 if not self.extrapolate[i,0]:
-                    lt_sel = x[i,:] < t[k]
-                    x[i,lt_sel] = t[k]
+                    lt_sel = x[:, i] < t[k]
+                    x[lt_sel, i] = t[k]
                 if not self.extrapolate[i,1]:
-                    gte_sel = t[-k-1] < x[i,:]
-                    x[i,gte_sel] = t[-k-1]
+                    gte_sel = t[-k-1] < x[:, i]
+                    x[gte_sel, i] = t[-k-1]
                 extrapolate_flag = True
 
 
-            self.impl.evaluate_spline(t, k, x[i,:], nu, extrapolate_flag, self.interval_workspace[i], self.basis_workspace[i],)
+            self.impl.evaluate_spline(t, k, x[:,i], nu, extrapolate_flag, self.interval_workspace[i], self.basis_workspace[i],)
             np.add(
-                self.coefficient_selector_base[i][..., None],
-                self.interval_workspace[i][:num_points],
-                out=self.coefficient_selector[i, ..., :num_points])
+                self.coefficient_selector_base[i][None, ...],
+                # Broadcasting does not play nciely with xdim as last axis for some reason,
+                # so broadcasting manually. Need to determine speed concequences.
+                self.interval_workspace[i][(slice(0,num_points),) + (None,)*self.xdim],
```

Comment:
@ixjlyons Does the azure pipeline run the benchmarks? Or is the best thing for me to run them on my own laptop? I'm curious about how to broadcast this most efficiently.

Reply:
Or if it matters.

Reply:
I did a bit of testing and I don't think it will matter too much

Reply:
Just for the record the benchmarks are only run when building the docs but no big deal to add a task to run them in CI as well.